File: CSC110-Cpp/Module14-File-IO/debug01/score_reader.cpp
// score_reader.cpp
// Reads student names and scores from scores.txt, prints each one, and
// reports how many students there are and the class average.
#include <iostream>
#include <fstream>
#include <string>
using namespace std;

int main() {
    ifstream in("scores.txt");
    if (!in.is_open()) {
        cout << "Could not open scores.txt" << endl;
        return 1;
    }

    string name;
    int score;
    int count = 0;
    int total = 0;

    while (!in.eof()) {
        in >> name >> score;
        cout << name << ": " << score << endl;
        total += score;
        count++;
    }
    in.close();

    cout << "Students: " << count << endl;
    cout << "Average: " << static_cast<double>(total) / count << endl;
    return 0;
}

File: CSC110-Java/Module14-File-IO/debug01/ScoreReader.java
// ScoreReader.java
// Reads exam scores, one per line, from scores.txt and prints the
// number of scores, the highest score, and the average.

import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class ScoreReader {
    public static void main(String[] args) {
        Scanner fileIn;
        try {
            fileIn = new Scanner(new File("scores.txt"));
        } catch (FileNotFoundException e) {
            System.out.println("Could not open scores.txt");
            return;
        }

        int count = 0;
        int total = 0;
        int highest = 0;

        while (fileIn.hasNextLine()) {
            int score = fileIn.nextInt();
            count++;
            total += score;
            if (score > highest) {
                highest = score;
            }
        }
        fileIn.close();

        System.out.println("Scores read: " + count);
        System.out.println("Highest:     " + highest);
        System.out.printf("Average:     %.2f%n", (double) total / count);
    }
}
